### DISCUSSION TITLE
Probleme beim Übergeben der signed challenge (Schritt 4)

### DISCUSSION THREAD

**Discussion by MirkoWe** [Digitaler-Impfnachweis/certification-apis#64] [Q&A] [answered]
Wir erhalten beim Übergeben der signed challenge folgende Rückantwort:
Wir verwenden den Endpoint 'https://keycloak-cert-auth.dev.ibmega.net/auth/realms/master/protocol/openid-connect/auth'
und testen mit dem Gematik KoPS 3.1 Konnektor Simulator.
Bei der challenge haben wir die Bindestriche entfernt, diese dann anschließend Base64 codiert und mit dem ExternalAuth Service signiert.
An was kann das liegen?

`<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" class="login-pf">

<head>
    <meta charset="utf-8">
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
    <meta name="robots" content="noindex, nofollow">

            <meta name="viewport" content="width=device-width,initial-scale=1"/>
    <title>Log in to Keycloak</title>
    <link rel="icon" href="/auth/resources/u2g3a/login/keycloak/img/favicon.ico" />
            <link href="/auth/resources/u2g3a/common/keycloak/node_modules/patternfly/dist/css/patternfly.min.css" rel="stylesheet" />
            <link href="/auth/resources/u2g3a/common/keycloak/node_modules/patternfly/dist/css/patternfly-additions.min.css" rel="stylesheet" />
            <link href="/auth/resources/u2g3a/common/keycloak/lib/zocial/zocial.css" rel="stylesheet" />
            <link href="/auth/resources/u2g3a/login/keycloak/css/login.css" rel="stylesheet" />
<SCRIPT> if (typeof history.replaceState === 'function') {  history.replaceState({}, "some title", "https://keycloak-cert-auth.dev.ibmega.net/auth/realms/master/login-actions/authenticate?execution=b088022e-4bca-4c84-97c8-011f3cad6257&client_id=smcb-auth-simulator&tab_id=1Wd-YcdvK-o"); }</SCRIPT></head>

<body class="">
  <div class="login-pf-page">
    <div id="kc-header" class="login-pf-page-header">
      <div id="kc-header-wrapper" class=""><div class="kc-logo-text"><span>Keycloak</span></div></div>
    </div>
    <div class="card-pf ">
      <header class="login-pf-header">
                <h1 id="kc-page-title">        We are sorry...
</h1>
      </header>
      <div id="kc-content">
        <div id="kc-content-wrapper">

        <div id="kc-error-message">
            <p class="instruction">Unexpected error during auth flow</p>
        </div>

        </div>
      </div>

    </div>
  </div>
</body>
</html>`

**Comment by mge-medisoftware** [comment, marked as answer]
In der TI RU funktioniert es ohne Entfernen der Bindestriche (Signaturvorgang mit secunet-Konnektor), zumindest zur Zeit.

**Reply by MirkoWe** [reply]
Mit dem secunet-Konnektor funktioniert es nun auch, allerdings bekomme ich bei Schritt 5 folgende Fehlermeldung: {"error":"RESTEASY003650: No resource method found for GET, return 405 with Allow header"}

Endpoint: https://keycloak-cert-auth.dev.ibmega.net/auth/realms/master/protocol/openid-connect/token

**Reply by oliver-steinbrecher** [reply]
Es scheint hier die falsche HTTP Methode genutzt zu werden.

**Reply by nicolaslang1** [reply]
Hallo MirkoWe,

wir nutzen auch KOPS 3.1. Zuerst haben wir die Challenge einfach direkt mit den Bindestrichen signiert und jetzt auch mal die Striche entfernt, da es bei Ihnen ja zu funktionieren schien. Dürfte ich fragen, mit welcher SMCB Ihr es probiert habt? Bekomme leider immernoch den Fehler: 
Keycloak

We are sorry... 

Certificate & Signature Verification failed

**Comment by ManuelB** [comment]
Ich hab mal unsere Lösung hier detailliert erklärt: https://youtu.be/oLrjF-OISO0 https://github.com/ere-health/dgc-ps-app
